src/components/OptionListContextProvider.tsx

```diff
@@ -108,14 +115,45 @@ function OptionsListContextProvider({reports, children}: OptionsListProviderProp
             return;
         }
 
+        const newReportOptions: Array<{
+            replaceIndex: number;
+            newReportOption: OptionsListUtils.SearchOption<Report>;
+        }> = [];
+
+        Object.keys(personalDetails).forEach((accoutID) => {
+            const prevPersonalDetail = prevPersonalDetails?.[accoutID];
+            const personalDetail = personalDetails?.[accoutID];
+
+            if (isEqualPersonalDetail(prevPersonalDetail, personalDetail)) {
+                return;
+            }
+
+            Object.values(reports ?? {})
+                .filter((report) => Boolean(report?.participantAccountIDs?.includes(Number(accoutID))) || (ReportUtils.isSelfDM(report) && report?.ownerAccountID === Number(accoutID)))
+                .forEach((report) => {
+                    if (!report) {
+                        return;
+                    }
+                    const newReportOption = OptionsListUtils.createOptionFromReport(report, personalDetails);
+                    const replaceIndex = options.reports.findIndex((option) => option.reportID === report.reportID);
+                    newReportOptions.push({
+                        newReportOption,
+                        replaceIndex,
+                    });
+                });
+        });
+
         // since personal details are not a collection, we need to recreate the whole list from scratch
         const newPersonalDetailsOptions = OptionsListUtils.createOptionList(personalDetails).personalDetails;
 
         setOptions((prevOptions) => {
             const newOptions = {...prevOptions};
             newOptions.personalDetails = newPersonalDetailsOptions;
+            newReportOptions.forEach((newReportOption) => (newOptions.reports[newReportOption.replaceIndex] = newReportOption.newReportOption));
             return newOptions;
         });
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
what other dependencies is this expecting that we're purposefully ignoring? it looks `reports`?

Reply:
@nkdengineer can you help me understand why we added this?

Reply:
@bondydaa In this case we only want to update the personal detail option and the related report of the changed personal detail. So we're ignoring other dependencies.

Reply:
little bump @nkdengineer on above [comment](https://github.com/Expensify/App/pull/40167#discussion_r1572811326)

Reply:
@bondydaa I added a comment, please help to check again.
cc @hungvu193

Reply:
little bump @bondydaa for final review